src/Plugin/Condition/DataListContains.php

```diff
@@ -39,6 +39,20 @@ class DataListContains extends RulesConditionBase {
   /**
    * {@inheritdoc}
    */
+  public function refineContextDefinitions() {
+
+    // Refine the item context.
+    if ($item = $this->getContextValue('item')) {
+      if ($item instanceof EntityInterface && $type = $item->getEntityTypeId()) {
```

Comment:
"$typed_data_item = $this->getContextData('item')" created this error:

PHP Fatal error:  Call to undefined method Drupal\rules\Plugin\Condition\DataListContains::getContextData() in /Users/steve/Sites/d8dev/modules/rules/src/Plugin/Condition/DataListContains.php on line 45

...so I copied the code from DataComparison.php:

$item = $this->condition->getContextDefinition('item');

Your suggested code failed my test though:

Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'any'
+'entity:entity_zero_type_id'